File: src/main/java/com/moovie/backend/domain/model/UserProfile.java
package com.moovie.backend.domain.model;

import java.util.List;

public record UserProfile(
        String photoUrl,
        String username,
        String bio,
        List<Movie> moviesWatched,
        List<UserSummary> following,
        List<UserSummary> followers
) {
}

File: src/main/kotlin/org/mobyle/domain/model/UserProfile.kt
package org.mobyle.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class UserProfile(
    val id: String = "",
    val photoUrl: String = "",
    val username: String,
    val bio: String = "",
    val moviesWatchedCount: Int = 0,
    val followingCount: Int = 0,
    val followersCount: Int = 0,
    val recentMovies: List<Movie> = emptyList()
)
